pipeline/gnomad/download_gnomad_threepointone.py

```diff
@@ -0,0 +1,406 @@
+                    ac
+                    an
+                    ac_hom
+                  }
+                }
+                flags
+                transcript_consequences {
+                  hgvsc
+                  lof
+                  lof_flags
+                  lof_filter
+                  major_consequence
+                  transcript_id
+                }
+              }
+            }
+      """
+    v_q = variant_query % (variant_id, dataset)
+    retries = 0
+    while retries < max_retries:
+        try:
+            # https://stackoverflow.com/questions/49064398/requests-exceptions-chunkedencodingerror-connection-broken-incompleteread0
+            response = requests.post(
+                'http://gnomad.broadinstitute.org/api',
+                json={ "query": v_q,
+                       "variables": {}},
+                headers={"content-type": "application/json"})
+            if response.ok:
+                time.sleep(0.1)
+                parse = json.loads(response.text)
+                print(f'success for variant {variant_id}')
+                return(parse['data']['variant'])
+            else:
+                print('rate limit reached, waiting 60 seconds...')
+                time.sleep(60)
+                continue
+        except json.decoder.JSONDecodeError:
+            logging.info(f'Request for variant {variant_id} failed, kicking off retry #{retries}')
+            retries += 1
+            time.sleep(3)
+            continue
+        except KeyError as e:
+            print(f"keyerror for variant {variant_id}, error is {e}")
+            retries += 1
+            time.sleep(300)
+            continue
+        except Exception as e:
+            print(f"unknown error: {e}, sleeping for a couple minutes...")
+            retries += 1
+            time.sleep(300)
+            continue
+    print(f"failed to get data for {variant_id}")
+    logging.info(f"failed to get data for {variant_id}")
+    return None
+    
+
+def gene_to_region_variants(gene, dataset, reference):
+    """
+    Given a gene name, return the list of variants via a region
+    query.  These will mostly be the intronic variants.
+    """
+    coords = gene_to_coords(gene, reference)
+    variantList = coords_to_variants(coords["chrom"], coords["start"],
+                                     coords["stop"], dataset, reference)
+    return variantList
+
+
+def getVariants(transcript, gene, dataset, reference):
+    exonic_set = set()
+    variant_ids = []
+    exonic_variants_non_cancer = transcript_to_variants(transcript, dataset, reference)
+    for variant in exonic_variants_non_cancer:
+        variant_id = variant['variant_id']
+        if variant_id not in variant_ids:
+            variant = fetch_data_for_one_variant(variant_id, dataset)
+            exonic_set.add(json.dumps(variant, sort_keys=True))
+            variant_ids.append(variant_id)
+    retries = 0
+    max_retries = 5
+    while retries < max_retries:
```

Comment:
I think it would have been nice to separate the retry logic from the querying code and the code processing the result of the query. It would be easier to follow the code